modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ddl/DdlCommandHandler.java

```diff
@@ -0,0 +1,281 @@
+import org.apache.ignite.schema.definition.builder.SortedIndexDefinitionBuilder;
+import org.apache.ignite.schema.definition.builder.SortedIndexDefinitionBuilder.SortedIndexColumnBuilder;
+
+/** DDL commands handler. */
+public class DdlCommandHandler {
+    private final TableManager tableManager;
+
+    public DdlCommandHandler(TableManager tblManager) {
+        tableManager = tblManager;
+    }
+
+    /** Handles ddl commands. */
+    public void handle(DdlCommand cmd, PlanningContext pctx) throws IgniteInternalCheckedException {
+        validateCommand(cmd);
+
+        if (cmd instanceof CreateTableCommand) {
+            handleCreateTable((CreateTableCommand) cmd);
+        } else if (cmd instanceof DropTableCommand) {
+            handleDropTable((DropTableCommand) cmd);
+        } else if (cmd instanceof AlterTableAddCommand) {
+            handleAlterAddColumn((AlterTableAddCommand) cmd);
+        } else if (cmd instanceof AlterTableDropCommand) {
+            handleAlterDropColumn((AlterTableDropCommand) cmd);
+        } else if (cmd instanceof CreateIndexCommand) {
+            handleCreateIndex((CreateIndexCommand) cmd);
+        } else if (cmd instanceof DropIndexCommand) {
+            handleDropIndex((DropIndexCommand) cmd);
+        } else {
+            throw new IgniteInternalCheckedException("Unsupported DDL operation ["
+                    + "cmdName=" + (cmd == null ? null : cmd.getClass().getSimpleName()) + "; "
+                    + "querySql=\"" + pctx.query() + "\"]");
+        }
+    }
+
+    /** Validate command. */
+    private void validateCommand(DdlCommand cmd) {
+        if (cmd instanceof AbstractDdlCommand) {
+            AbstractDdlCommand cmd0 = (AbstractDdlCommand) cmd;
+
+            if (IgniteUtils.nullOrEmpty(cmd0.tableName())) {
+                throw new IllegalArgumentException("Table name is undefined.");
+            }
+        }
+    }
+
+    /** Handles create table command. */
+    private void handleCreateTable(CreateTableCommand cmd) {
+        PrimaryKeyDefinitionBuilder pkeyDef = SchemaBuilders.primaryKey();
+        pkeyDef.withColumns(cmd.primaryKeyColumns());
+        pkeyDef.withAffinityColumns(cmd.affColumns());
+
+        Consumer<TableChange> tblChanger = tblCh -> {
+            TableChange conv = convert(SchemaBuilders.tableBuilder(cmd.schemaName(), cmd.tableName())
+                    .columns(cmd.columns())
+                    .withPrimaryKey(pkeyDef.build()).build(), tblCh);
+
+            if (cmd.partitions() != null) {
+                conv.changePartitions(cmd.partitions());
+            }
+
+            if (cmd.replicas() != null) {
+                conv.changeReplicas(cmd.replicas());
+            }
+        };
+
+        String fullName = TableDefinitionImpl.canonicalName(cmd.schemaName(), cmd.tableName());
+
+        if (cmd.ifTableExists()) {
+            tableManager.createTableIfNotExists(fullName, tblChanger);
```

Comment:
`if table exists then create table if not exists` -- this one looks very strange. Let's rename flag to match sql statement (`IF NOT EXISTS`)